### ISSUE TITLE
Stabilisation of the connection handling for p2p and HCS clients (retrying)

### ISSUE THREAD

**Issue by gregscullard** [LimeChain/HCS-Integration-Node#22]
As the code stands, if the connection to the mirror is lost, the subscription will end. It should attempt to reconnect after a few seconds' delay.

**Comment by Perseverance** [comment]
Hey @gregscullard . Yeah, stabilization work is something that we want to tackle once we are done with the client side apps for the demo. Would you mind me extending this issue to include not only mirror node connection retries, but also the p2p client, as the p2p nodes retries would require similar work too?

**Comment by gregscullard** [comment]
Agreed, although I'm less familiar with p2p frameworks to be able to comment.
